src/dl_typelib_write_c_header.cpp

```diff
@@ -113,6 +113,86 @@ static void dl_context_write_c_header_begin( dl_binary_writer* writer, const cha
 									   "#  if !defined(DL_STATIC_ASSERT)\n"
 									   "#    define DL_STATIC_ASSERT(x,y) // default to non-implemented.\n"
 									   "#  endif\n"
+									   "\n"
+									   "   // ... DL_ASSERT() ...\n"
+									   "#  if !defined(DL_ASSERT)\n"
+									   "#    define DL_ASSERT(x) // default to non-implemented.\n"
+									   "#  endif\n"
+									   "\n"
+									   "   // ... DLArray ...\n"
+									   "#  if defined(__cplusplus)\n"
+									   "	  template <typename T>\n"
+									   "	  struct DLArrayStruct\n"
```

Comment:
I know naming is DL is not unified across the codebase and that is a problem, but the public api is always lower_case_underscore.
I would go with 'dl_array' I think...